docs/src/main/asciidoc/spring-cloud-gateway.adoc

```diff
@@ -1553,6 +1553,65 @@ route.getMetadata();
 route.getMetadata(someKey);
 ```
 
+== Http timeouts configuration
+
+Http timeouts (response and connect) can be configured for all routes and overridden for each specific route.
+
+=== Global timeouts
+
+.application.yml
+[source,yaml]
+----
+spring:
+  cloud:
+    gateway:
+      httpclient:
+        connect-timeout: 1000
+        response-timeout: 5s
+----
+
+=== Per-route timeouts
+NOTE: response and connect timeout must be specified in milliseconds.
```

Comment:
why does the `response-timeout` have an `s`, should it not be `5000`?

Reply:
`spring.cloud.gateway.httpclient.response-timeout` property has type Duration
https://github.com/spring-cloud/spring-cloud-gateway/blob/master/spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/config/HttpClientProperties.java#L54

while per-route configuration stored in metadata Map<String, Object> and since this map is untyped I decided to store Long value there

Reply:
This `NOTE` is confusing then.  So it is just the metadata that needs to be specified only in milliseconds?  If the global timeouts can use different `Duration` types then we should change this `NOTE` to reflect that.

Reply:
I reworked documentation to make it clearer how a user should configure HTTP timeouts.